File: java/metodos/funcion_para_buscar_contacto.java
package metodos;
import java.util.Scanner;
public class funcion_para_buscar_contacto {
    public static boolean buscarContacto(String[] contactos, String nombreBuscado) {
        for (int i = 0; i < contactos.length; i++) {
            if (contactos[i].equalsIgnoreCase(nombreBuscado)) {
                return true;
            }
        }
        return false;
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] contactos = {"Juan", "Maria", "Carlos", "Ana"};
        System.out.print("Ingrese el nombre a buscar: ");
        String nombre = scanner.nextLine();
        boolean encontrado = buscarContacto(contactos, nombre);
        if (encontrado) {
            System.out.println("Contacto encontrado en la agenda.");
        } else {
            System.out.println("Contacto no existe.");
        }
        scanner.close();
    }
}

File: visual_basic/metodos/funcion_para_buscar_contacto/funcion_para_buscar_contacto.vb
Imports System
Module funcion_para_buscar_contacto
    Sub Main(args As String())
        Dim contactos As New List(Of String) From {
            "Juan", "Maria", "Carlos", "Ana"
        }
        BuscarContacto(contactos)
        Console.ReadKey()
    End Sub
    Sub BuscarContacto(ByVal lista As List(Of String))
        Console.WriteLine("Ingrese el nombre del contacto a buscar:")
        Dim nombre As String = Console.ReadLine()
        Dim encontrado As Boolean = False
        For Each contacto As String In lista
            If contacto.ToLower() = nombre.ToLower() Then
                encontrado = True
                Exit For
            End If
        Next
        If encontrado Then
            Console.WriteLine("✅ Contacto encontrado")
        Else
            Console.WriteLine("❌ Contacto no encontrado")
        End If
    End Sub
End Module